dartagnan/src/main/java/com/dat3m/dartagnan/witness/graphml/WitnessBuilder.java

```diff
@@ -188,20 +189,20 @@ public WitnessGraph build() {
         return graph;
     }
 
-    private List<Event> getSCExecutionOrder(Model model) {
+    private List<Event> getSCExecutionOrder(IREvaluator model) {
         // TODO: we recently added many cline to many events and this might affect the
         // witness generation.
-        Predicate<Event> executedCEvents = e -> Boolean.TRUE.equals(model.evaluate(context.execution(e)))
-                && e.hasMetadata(SourceLocation.class);
-        List<Event> execEvents = context.getTask().getProgram().getThreadEvents().stream().filter(executedCEvents)
+        Predicate<Event> executedCEvents = e -> model.isExecuted(e) && e.hasMetadata(SourceLocation.class);
+        List<Event> execEvents = context.getTask().getProgram().getThreadEvents().stream()
+                .filter(executedCEvents)
                 .toList();
         Map<Integer, List<Event>> map = new HashMap<>();
         for (Event e : execEvents) {
             // TODO improve this: these events correspond to return statements
             if (e instanceof Store store && store.getMemValue() instanceof BoolLiteral bVal && !bVal.getValue()) {
                 continue;
             }
-            BigInteger var = model.evaluate(context.clockVariable("hb", e));
+            BigInteger var = model.getSMTModel().evaluate(context.clockVariable("hb", e));
```

Comment:
Create "IREvaluator.happensBeforeClock()" method similar to `IREvaluator..memoryOrderClock()`

Reply:
No, IREvaluator should not have such specific functionality that only makes sense for one particular use-case related to SVCOMP. It could have a general `clockValue(Relation rel, Event e)` method though (but it's also a bit questionable).